File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// defining my functions
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // asking the user for the text
    string text = get_string("Text: ");
    // getting letters , words and sentecnes
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    // setting the variables
    float L = letters / (float) words * 100;
    float S = sentences / (float) words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int X = round(index);
    // printing the results
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", X);
    }

}

//counting the letters
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] >= 'A' && text[i] <= 'Z')
        {
            letters++;
        }
        else if (text[i] >= 'a' && text[i] <= 'z')
        {
            letters++;
        }
    }
    return letters;
}

// counting the words (words = spaces +1)
int count_words(string text)
{
    int spaces = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            spaces++;
        }
    }
    int words = spaces + 1;
    return words;
}

// counting the sentences that end with '.' or '?' or '!'
int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
from cs50 import get_string
# my variables
letters = 0
spaces = 0
words = 0
sentences = 0
# getting input (text) from the user
text = get_string("Text: ")

for i in range(len(text)):
    if text[i].isupper() or text[i].islower():  # for letters
        letters += 1
    elif text[i] == " ":  # for words
        spaces += 1
        words = spaces + 1
    elif text[i] == '.' or text[i] == '!' or text[i] == '?':
        sentences += 1

L = letters / words * 100
S = sentences / words * 100
index = 0.0588 * L - 0.296 * S - 15.8
grade = round(index)

if index < 1:
    print("Before Grade 1")
elif index > 1 and index < 16:
    print(f"Grade {grade}")
elif index > 16:
    print("Grade 16+")